src/backtrace/mod.rs

```diff
@@ -37,10 +36,20 @@ use std::os::raw::c_void;
 /// }
 /// ```
 #[inline(never)]
+#[cfg(feature = "std")]
 pub fn trace<F: FnMut(&Frame) -> bool>(mut cb: F) {
     trace_imp(&mut cb)
 }
 
+/// Without std this function now does not have synchronization guarentees.
+/// Please refer to std documentation for examples and explaination.
+#[inline(never)]
+#[cfg(not(feature = "std"))]
+pub unsafe fn trace<F: FnMut(&Frame) -> bool>(mut cb: F) {
```

Comment:
Similar comments to `resolve` below in that we won't want to switch signatures based on features, but rather just not provide the safe API in `std`